src/api_v2/api.rs

```diff
@@ -0,0 +1,143 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use serde_json::json;
+use crate::api_v2::{Error, User};
+
+#[derive(Default)]
+pub struct Scopes {
+    str: String,
+}
+
+impl Scopes {
+    pub fn new() -> Self {
+        return Self::default();
+    }
+
+    /// Allows sending chat messages on a user's behalf.
+    ///
+    /// Using the chat.write scope requires either:
+    /// - a Chat Bot account to send messages on behalf of other users.
+    /// - Authorization code grant where the user is the same as the client's owner (send as yourself).
+    pub fn chat(self) -> Self {
+        return self.add_scope("chat.write");
+    }
+
+    /// Allows acting as the owner of a client; only available for Client Credentials Grant.
+    pub fn delegate(self) -> Self {
+        return self.add_scope("delegate");
+    }
+
+    /// Allows creating and editing forum posts on a user's behalf.
+    pub fn forum(self) -> Self {
+        return self.add_scope("forum.write");
+    }
+
+    /// Allows reading of the user's friend list.
+    pub fn friends(self) -> Self {
+        return self.add_scope("friends.read");
+    }
+
+    ///
+    /// Allows reading of the public profile of the user (/me).
+    ///
+    /// `identify`  is the default scope for the Authorization Code Grant and always implicitly provided.
+    pub fn identify(self) -> Self {
+        return self.add_scope("identify");
+    }
+
+    /// Allows reading of publicly available data on behalf of the user.
+    pub fn public(self) -> Self {
+        return self.add_scope("public");
+    }
+
+    fn add_scope(mut self, scope: &str) -> Self {
+        let s = &self.str;
+        self.str = format!("{s} {scope}");
```

Comment:
Instead of concat string yourself, try to implement this by `Vec` and give a new method `build()` to call `Vec::join`  after process done.